### PR TITLE
feature: add market price diff percentage

### PR DESCRIPTION
**Pull request by berteotti**
# Summary

Adds market price difference in percentage in the limit order price

image

  # To Test

- [ ] Check if there is a market price difference in percentage in the limit order price
- [ ] Test if the difference is correct

### CODE REVIEW

File: src/pages/Swap/LimitOrderBox/components/LimitOrderForm/LimitOrderForm.tsx [hunk 1]
```diff
@@ -391,6 +391,50 @@ export function LimitOrderForm({ account, provider, chainId }: LimitOrderFormPro
...
+
+      if (cowQuote) {
+        const {
+          quote: { buyAmount, sellAmount },
+        } = cowQuote
+
+        if (limitOrder.kind === LimitOrderKind.SELL)
+          setLimitOrderPrices(limitPrice => ({
+            ...limitPrice,
+            buy: parseFloat(formatUnits(buyAmount, buyTokenAmount.currency.decimals) ?? 0) / 10,
```
Comment (Diogomartf) [inline src/pages/Swap/LimitOrderBox/components/LimitOrderForm/LimitOrderForm.tsx:419]: this code is very similar to the one bellow, maybe an opportunity to make a function?

File: src/pages/Swap/LimitOrderBox/components/OrderLimitPriceField/OrderLimitPriceField.tsx [hunk 2]
```diff
@@ -35,6 +43,33 @@ export function OrderLimitPriceField({ id }: OrderLimitPriceFieldProps) {
   const toggleCurrencyButtonLabel = `${quoteTokenAmount?.currency?.symbol}`
   const [inputLimitPrice, setInputLimitPrice] = useState<string | number>(formattedLimitPrice)
 
+  const nextLimitPriceFloat = limitOrder.kind === LimitOrderKind.SELL ? limitOrderPrices.buy : limitOrderPrices.sell
+
+  console.log(limitOrder.kind, nextLimitPriceFloat.toFixed(6), formattedLimitPrice)
```
Comment (Diogomartf) [inline src/pages/Swap/LimitOrderBox/components/OrderLimitPriceField/OrderLimitPriceField.tsx:48]: remove log?

File: src/pages/Swap/LimitOrderBox/components/OrderLimitPriceField/styles.ts [hunk 3]
```diff
@@ -50,3 +50,7 @@ export const ToggleCurrencyButton = styled.span`
   color: #464366;
   cursor: pointer;
 `
+
+export const MarketPriceDiff = styled.span<{ isPositive: boolean }>`
+  color: ${({ isPositive }) => (isPositive ? 'green' : 'red')};
```
Comment (Diogomartf) [inline src/pages/Swap/LimitOrderBox/components/OrderLimitPriceField/styles.ts:55]: this misses the case when there is no price difference, maybe we want to show gray on that case.

Note: we have that on SimpleChart, would be great to start to reuse logic

Comment (berteotti) [reply src/pages/Swap/LimitOrderBox/components/OrderLimitPriceField/styles.ts:55]: If the price has no difference we don't show it. So no need for a condition that isn't met

File: src/pages/Swap/LimitOrderBox/contexts/LimitOrderFormContext.tsx [hunk 4]
```diff
@@ -49,6 +50,11 @@ interface ILimitOrderFormContext {
   expiresInUnit: OrderExpiresInUnit
   setExpiresInUnit: (expiresInUnit: OrderExpiresInUnit) => void
   setToMarket: () => void
+  /**
+   *  market price difference
+   */
```
[the remark this answers is not in the document]
Comment (berteotti) [reply src/pages/Swap/LimitOrderBox/contexts/LimitOrderFormContext.tsx:55]: I think it can go for now to have the same comments as above.

**Pull request merged**